### ISSUE TITLE
CRC reported by ZipFile::crc32() seems to be the actual CRC of compressed file, instead of the one saved in ZIP file structure

### ISSUE THREAD

**Issue by michalrud** [mvdnes/zip-rs#24]
Hi,

First of most, thanks for creating this library, using it is a joy 😄 

I've stumbled upon a problem and I am not sure whether I am doing something wrong, or there is a problem in a library. Part of my application's functionality would be to check whether a given ZIP file is damaged; I wanted to do this by reading a CRC32 from a file structure and then compare it with the CRC32 I've calculated from the file which was uncompressed to some temporary directory. To check if it works okay, I've modified an existing, small ZIP file in a hex editor and changed value of CRC32 to an incorrect value by hand. [You can find this file here](https://github.com/michalrud/zip-crc-example/blob/master/testfiles/wrongcrc.zip). As expected, `zip` tool sees it as an invalid zip file:

```
$ zip -T wrongcrc.zip
hello.txt               bad CRC 4a92ba89  (should be 13371337)
test of wrongcrc.zip FAILED

zip error: Zip file invalid, could not spawn unzip, or wrong unzip (original files unmodified)
```

However, trying to use the `crc32()` method from a `ZipFile` object gives me the correct CRC32, instead of the wrong one that is present in this file. Minimal example showing this issue can be found in the following repository:
https://github.com/michalrud/zip-crc-example

Am I using the library wrong? Maybe I misunderstood the documentation? Thanks in advance for any help :)

**Comment by mvdnes** [comment]
Thanks for the kind words, and for this very detailed issue.

A Zip file has two locations to store the file metadata, such as the checksum.
One place is the Local file header, which contains the CRC you changed.
The other place is at the end of a Zip file, in the Central Directory. This is where my library gets most of it's information, including the checksum. If you modify the checksum there (in your file at 0x6e-0x71), it will actually report them as incorrect.

I know that the library should actually compare the information stored in the local and central file header, but had not thought of an idea to cope with the errors.

So to recap: no actual recomputation is done, it just looks for the value in another place.

**Comment by michalrud** [comment]
Thanks a lot, that explains everything, I guess issue can be closed.
